cli/module_registry.go

```diff
@@ -316,6 +332,30 @@ func UploadModuleAction(c *cli.Context) error {
 	return nil
 }
 
+// call validateModelAPI on all models in manifest and warn if violations.
+func validateModels(errWriter io.Writer, manifest *moduleManifest) {
+	for _, model := range manifest.Models {
+		if err := validateModelAPI(model.API); err != nil {
+			warningf(errWriter, "error validating API string %s: %s", model.API, err)
+		}
+	}
+}
+
+// return a useful error if the model string looks wrong.
+func validateModelAPI(modelAPI string) error {
+	api, err := resource.ParseAPIString(modelAPI)
+	if err != nil {
+		return errors.Wrap(err, "unparseable model string")
+	}
+	if err := api.Validate(); err != nil {
+		return errors.Wrap(err, "failed to validate API")
+	}
+	if api.Type.Namespace == resource.APINamespaceRDK && !slices.Contains(rdkAPITypes, api.Type.Name) {
```

Comment:
Looking through the code as-is in robot/impl.resource_manager.go (loc 650) and module/modmanager/manager.go (loc 1157) (switch in registerResources), I think we require that all resources are components or services (even if they are non-rdk namespaced apis (ex: other:unknown:x is invalid but other:component:x is valid))

Reply:
(Found by searching for `IsService`)

Reply:
Yeah good call out; even custom APIs must use a type of either `service` or `component` IIUC. I think you can remove the namespace-check part of this predicate. FWIW there's one other type: `remote` and is used to store remotes in the resource graph, but it should only be available internally.